### PR TITLE
[CI] add github action for pytest

### PR DESCRIPTION
**Pull request by jonasjucker**
Move CI entirely to GitHub actions

Add coverage measurement (no effect yet)

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[run]
+source =
+	engine/
+    probtest.py
+	util/
```
Comment (huppd) [inline .coveragerc:5]: Should we add visualize/ as well here? I know it might be bad for the coverage.

File: jenkins/mergerequest
Comment (huppd) [inline jenkins/mergerequest:1]: I would like to keep it for know, even though it is redundant. But we are currently considering different options to test on CSCS and this is like one example.

**Review by huppd — APPROVED** [review]
It already looks great to me. This is very good step forward. I like that we now also have coverage tests. Just some minor suggestions.

File: .coveragerc (hunk 1 above)
Comment (jonasjucker) [reply .coveragerc:6]: I added it. Makes sense to check this as well.

File: jenkins/mergerequest
Comment (jonasjucker) [reply jenkins/mergerequest:1]: Restored. Let's keep it for now.

Comment (huppd) [reply jenkins/mergerequest:1]: thanks

**Review by huppd — APPROVED** [review]
approved!

**Pull request merged**